pkg/extensions/sync/references/cosign.go

```diff
@@ -176,45 +219,50 @@ func (ref CosignReference) SyncReferences(localRepo, remoteRepo, subjectDigestSt
 					SignatureDigest: referenceDigest.String(),
 				})
 			} else {
-				err = repodb.SetImageMetaFromInput(localRepo, cosignTag, manifest.MediaType,
+				err = repodb.SetImageMetaFromInput(localRepo, cosignTag, ispec.MediaTypeImageManifest,
 					referenceDigest, manifestBuf, ref.storeController.GetImageStore(localRepo),
 					ref.repoDB, ref.log)
 			}
 
 			if err != nil {
-				return fmt.Errorf("failed to set metadata for cosign reference in '%s@%s': %w", localRepo, subjectDigestStr, err)
+				return refsDigests, fmt.Errorf("failed to set metadata for cosign reference in '%s@%s': %w",
+					localRepo, subjectDigestStr, err)
 			}
 
 			ref.log.Info().Str("repository", localRepo).Str("subject", subjectDigestStr).
 				Msg("repoDB: successfully added cosign reference for image")
 		}
 	}
 
-	return nil
+	return refsDigests, nil
 }
 
-func (ref CosignReference) getManifest(repo, cosignTag string) (*ispec.Manifest, error) {
-	var cosignManifest ispec.Manifest
+func (ref CosignReference) getManifest(repo, cosignTag string) (*v1.Manifest, godigest.Digest, error) {
+	/* using the same manifest type as cosign package,
+	the difference is in the order of json fields ("size" before "digest"), so digests will differ if we are using
+	the dist spec Manifest struct.
+	*/
+	var cosignManifest v1.Manifest
```

Comment:
using the same struct as cosign ensures that synced cosign references are identical to the upstream ones, otherwise digests will differ and will make this logic more complicated.

thoughts?